malaria24/ona/models.py

```diff
@@ -157,14 +165,25 @@ def get_digest_email_data(self):
 
                 if province_cases:
                     start_date = province_cases \
-                        .first().create_date_time.strftime(
-                            "%d %B %Y"
-                        )
+                        .first().create_date_time
+                    if min_date == temp_min_date:
+                        min_date = start_date
+                    else:
+                        if start_date < min_date:
+                            min_date = start_date
+
                     end_date = province_cases \
-                        .last().create_date_time.strftime(
-                            "%d %B %Y"
-                        )
-                    week = "{0} to {1}".format(start_date, end_date)
+                        .last().create_date_time
+                    if max_date == temp_max_date:
+                        max_date = end_date
+                    else:
+                        if end_date > max_date:
+                            max_date = end_date
+                    week = "{0} to {1}".format(min_date.strftime(
+                        "%d %B %Y"
+                    ), max_date.strftime(
+                        "%d %B %Y"
+                    ))
```

Comment:
This is for the date on each of the province objects, so I think here we want `start_date` and `end_date`.

And then for `week` on line 224, we want to use `min_date` and `max_date` (I think line 224 is where the actual bug is).